src/main/java/com/appdynamics/extensions/webspheremq/WMQMonitorTask.java

```diff
@@ -134,196 +149,248 @@ private PCFMessageAgent initPCFMesageAgent(MQQueueManager ibmQueueManager) {
-        String cmd =
-            wmqOverride.getIbmCommand() == null
-                ? "MQCMD_INQUIRE_Q_MGR_STATUS"
-                : wmqOverride.getIbmCommand();
-        metricsByCommand.putIfAbsent(cmd, new HashMap<>());
-        metricsByCommand.get(cmd).put(key, wmqOverride);
-      }
-      Map<String, WMQMetricOverride> queueManagerMetricsToReport =
-          metricsByCommand.get("MQCMD_INQUIRE_Q_MGR_STATUS");
-      if (queueManagerMetricsToReport != null) {
-        MetricCreator metricCreator =
-            new MetricCreator(monitorContextConfig.getMetricPrefix(), queueManager);
-        MetricsCollectorContext context =
-            new MetricsCollectorContext(
-                queueManagerMetricsToReport, queueManager, agent, metricWriteHelper);
-        MetricsPublisher qMgrMetricsCollector =
-            new QueueManagerMetricsCollector(context, metricCreator);
-        Runnable job = new MetricsPublisherJob(qMgrMetricsCollector, countDownLatch);
-        monitorContextConfig
-            .getContext()
-            .getExecutorService()
-            .execute("QueueManagerMetricsCollector", job);
-      }
-      Map<String, WMQMetricOverride> inquireMetricsToReport =
-          metricsByCommand.get("MQCMD_INQUIRE_Q_MGR");
-      if (inquireMetricsToReport != null) {
-        MetricCreator metricCreator =
-            new MetricCreator(
-                monitorContextConfig.getMetricPrefix(),
-                queueManager,
-                InquireQueueManagerCmdCollector.ARTIFACT);
-        MetricsCollectorContext context =
-            new MetricsCollectorContext(
-                inquireMetricsToReport, queueManager, agent, metricWriteHelper);
-        MetricsPublisher inquireQueueMgrMetricsCollector =
-            new InquireQueueManagerCmdCollector(context, metricCreator);
-        Runnable job = new MetricsPublisherJob(inquireQueueMgrMetricsCollector, countDownLatch);
-        monitorContextConfig
-            .getContext()
-            .getExecutorService()
-            .execute("InquireQueueManagerCmdCollector", job);
-      }
-    } else {
-      logger.warn("No queue manager metrics to report");
+    // Step 2: Inquire each metric type
+    inquireQueueMangerMetrics(
+        metricsMap.get(Constants.METRIC_TYPE_QUEUE_MANAGER), countDownLatch, mqQueueManager, agent);
+
+    inquireChannelMetrics(
+        metricsMap.get(Constants.METRIC_TYPE_CHANNEL), countDownLatch, mqQueueManager, agent);
+
+    inquireQueueMetrics(
+        metricsMap.get(Constants.METRIC_TYPE_QUEUE),
+        QueueCollectorSharedState.getInstance(),
+        countDownLatch,
+        agent);
+
+    inquireListenerMetrics(
+        metricsMap.get(Constants.METRIC_TYPE_LISTENER),
+        ListenerMetricsCollector::new,
+        ListenerMetricsCollector.ARTIFACT,
+        countDownLatch,
+        agent);
+
+    inquireTopicMetrics(
+        metricsMap.get(Constants.METRIC_TYPE_TOPIC),
+        TopicMetricsCollector::new,
+        null,
```

Comment:
probably doesn't make sense to pass the null here...but it does point out something weirdly inconsistent with regard to the use of artifact there. Hmm.....

Reply:
yes, somehow this type does not require the artifact name.    but sure, let me not pass in the null explicitly.